indico/modules/events/contributions/controllers.py

```diff
@@ -206,3 +237,47 @@ def _get_defaults(self):
         acl = {p.principal for p in self.contrib.acl_entries if p.read_access}
         return {'managers': managers, 'submitters': submitters, 'protection_mode': self.contrib.protection_mode,
                 'acl': acl}
+
+
+class RHContributionSubContributions(RHManageContributionBase):
+    """Get a list of subcontributions"""
+
+    def _process(self):
+        return jsonify_data(html=_render_subcontribution_list(self.contrib))
+
+
+class RHCreateSubContribution(RHManageContributionBase):
+    """Create a subcontribution"""
+
+    def _process(self):
+        form = SubContributionForm(event=self.event_new)
+        if form.validate_on_submit():
+            create_subcontribution(self.contrib, form.data)
+            return jsonify_data(html=_render_subcontribution_list(self.contrib))
```

Comment:
Don't you need to flash anything before returning? I don't know how the UI will look like, so it's just a question :)